packages/client-react-streaming/src/ReadableStreamLink.tsx

```diff
@@ -0,0 +1,137 @@
+import { ApolloLink, Observable } from "@apollo/client/index.js";
+import type { FetchResult } from "@apollo/client/index.js";
+
+export type ReadableStreamLinkEvent =
+  | { type: "next"; value: FetchResult }
+  | { type: "completed" }
+  | { type: "error" };
+
+interface InternalContext {
+  [teeToReadableStreamKey]?: ReadableStreamDefaultController<ReadableStreamLinkEvent>;
+  [readFromReadableStreamKey]?: ReadableStream<ReadableStreamLinkEvent>;
+}
+
+const teeToReadableStreamKey = Symbol.for(
+  "apollo.tee.readableStreamController"
+);
+const readFromReadableStreamKey = Symbol.for("apollo.read.readableStream");
+
+/**
+ * Apply to a context that will be passed to a link chain containing `TeeToReadableStreamLink`.
+ * @param controller
+ * @param context
+ * @returns
+ */
+export function teeToReadableStream<T extends Record<string, any>>(
+  controller: ReadableStreamDefaultController<ReadableStreamLinkEvent>,
+  context: T
+): T & InternalContext {
+  return Object.assign(context, {
+    [teeToReadableStreamKey]: controller,
+  });
+}
+
+/**
+ * Apply to a context that will be passed to a link chain containing `ReadFromReadableStreamLink`.
+ * @param readableStream
+ * @param context
+ * @returns
+ */
+export function readFromReadableStream<T extends Record<string, any>>(
```

Comment:
Same question as the link. Should these be marked as `@internal` since they seem to be used internally in #389?